src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/optimizer/rewriting/RuleApplicationBaseImpl.java

```diff
@@ -0,0 +1,75 @@
+package se.liu.ida.hefquin.engine.queryproc.impl.optimizer.rewriting;
+
+import se.liu.ida.hefquin.engine.queryplan.PhysicalPlan;
+import se.liu.ida.hefquin.engine.queryplan.physical.BinaryPhysicalOp;
+import se.liu.ida.hefquin.engine.queryplan.physical.NaryPhysicalOp;
+import se.liu.ida.hefquin.engine.queryplan.physical.UnaryPhysicalOp;
+import se.liu.ida.hefquin.engine.queryplan.physical.impl.PhysicalPlanWithBinaryRootImpl;
+import se.liu.ida.hefquin.engine.queryplan.physical.impl.PhysicalPlanWithNaryRootImpl;
+import se.liu.ida.hefquin.engine.queryplan.physical.impl.PhysicalPlanWithUnaryRootImpl;
+
+public abstract class RuleApplicationBaseImpl implements RuleApplication{
+    protected final PhysicalPlan[] subPlans;
```

Comment:
The meaning and the purpose of this member should be made clearer by renaming it to: `pathToTargetSubPlan`.